packages/grafana-runtime/src/services/templateSrv.ts

```diff
@@ -32,6 +32,16 @@ export interface TemplateSrv {
    * Update the current time range to be used when interpolating __from / __to variables.
    */
   updateTimeRange(timeRange: TimeRange): void;
+
+  /**
+   * Lists ad-hoc filters for a specific datasource
+   */
+  getAdhocFilters(datasourceName: string): AdHocVariableFilter[];
+
+  /**
+   * Extracts variable name from an expression like '${example}'
+   */
+  getVariableName(expression: string): string | undefined | null;
```

Comment:
```suggestion
  getVariableName(expression: string): string | null;
```

What would be the use case for the `undefined` return value of this function? I mean, having 2 different falsy return types looks confusing at first glance.

Reply:
That's perfect, thank you!